### PR TITLE
Add content describing contributions from outside of LLNL org

### PR DESCRIPTION
**Pull request by rhornung67**
# Summary

- This PR adds some documentation to assist contributors outside of the LLNL GitHub org.

It resolves https://github.com/LLNL/axom/issues/1241

### CODE REVIEW

File: src/docs/sphinx/dev_guide/dev_summary.rst [hunk 1]
```diff
@@ -72,6 +72,97 @@ See :ref:`pullrequest-label` for a description of our review process and
...
+Forking the repository
+--------------------------
+
+The requirement for individuals outside of the LLNL GitHub organization
+to contribute on a fork of the repo is due to policies enforced
+by the LLNL organization on GitHub (in which the Axom project resides) and the
+Livermore Computing (LC) organization (in which we run our GitLab CI testing).
+Fortunately, you may still contribute to Axom by `forking the Axom repo
+<https://github.com/LLNL/axom/fork>`_. Forking creates a copy of the Axom
+repository that you own. You can make changes on your local copy and push them
```
Comment (gunney1) [inline src/docs/sphinx/dev_guide/dev_summary.rst:114]: Missing a "to", "push them _to_ your fork on GitHub"

File: src/docs/sphinx/dev_guide/dev_summary.rst [hunk 2]
```diff
@@ -72,6 +72,97 @@ See :ref:`pullrequest-label` for a description of our review process and
...
+your fork on GitHub. When you are ready to have your Axom contribution reviewed
+ad added to the Axom project, you may create a pull request in the Axom project.
```
Comment (gunney1) [inline src/docs/sphinx/dev_guide/dev_summary.rst:116]: ad -> and

**Review by gunney1 — APPROVED** [review]
Just a couple of typos.  Thanks @rhornung67

**Review by agcapps — APPROVED** [review]
Thanks, Rich.

**Review by kennyweiss — APPROVED** [review]
Thanks @rhornung67 !

File: src/docs/sphinx/dev_guide/dev_summary.rst (hunk 1 above)
Comment (rhornung67) [reply src/docs/sphinx/dev_guide/dev_summary.rst:114]: Fixed.

File: src/docs/sphinx/dev_guide/dev_summary.rst (hunk 2 above)
Comment (rhornung67) [reply src/docs/sphinx/dev_guide/dev_summary.rst:116]: Fixed.

**Pull request merged**